src/lib/components/namespace-list.svelte

```diff
@@ -3,99 +3,155 @@
+        return !element.disabled;
+      });
+    }
+    return [];
+  };
+
+  const handleTabKey = (event: KeyboardEvent) => {
+    const focusable = getFocusableElements();
+    const inputElement = focusable[0];
+    const lastListItemElement = focusable[focusable.length - 1];
+    if (event.key === 'Tab') {
+      if (event.shiftKey) {
+        if (document.activeElement === inputElement) {
+          lastListItemElement.focus();
+          focusedIndex = focusable.length - 1;
+          event.preventDefault();
+        } else {
+          focusedIndex = focusedIndex - 1;
+        }
+      } else {
+        if (document.activeElement === lastListItemElement) {
+          inputElement.focus();
+          focusedIndex = 0;
+          event.preventDefault();
+        } else {
+          focusedIndex = focusedIndex + 1;
         }
       }
-    };
+    }
+  };
 
-    document.addEventListener('keydown', handleKeyDown, true);
+  const handleArrowKey = (event: KeyboardEvent) => {
+    const focusable = getFocusableElements();
+    if (event.key === 'ArrowUp') {
+      event.preventDefault();
+      if (focusedIndex === 1) return;
+      else {
+        const next = focusable[focusedIndex - 1];
+        if (next) {
+          next.focus();
+          focusedIndex = focusedIndex - 1;
+        }
+      }
+    } else if (event.key === 'ArrowDown') {
+      event.preventDefault();
+      if (focusedIndex === 0) {
+        // Input tab prevents updating focusedIndex
+        const next = focusable[2];
+        if (next) {
+          next.focus();
+          focusedIndex = focusedIndex + 2;
+        }
+      } else {
+        const next = focusable[focusedIndex + 1];
+        if (next) {
+          next.focus();
+          focusedIndex = focusedIndex + 1;
+        }
+      }
+    }
+  };
 
-    return {
-      destroy() {
-        document.removeEventListener('keydown', handleKeyDown, true);
-      },
-    };
-  }
+  const handleKeyboardNavigation = (event: KeyboardEvent) => {
+    const focusable = getFocusableElements();
+    if (!show || !focusable.length || focusable.length === 1) {
+      return;
+    }
 
-  $: searchValue = '';
-</script>
+    handleTabKey(event);
+    handleArrowKey(event);
+  };
 
-<div class="prose mt-16 mb-8">
-  <h2 class="text-2xl" data-testid="namespace-select-header">
-    Select a Namespace
-  </h2>
-  {#if $page.params?.namespace}
-    <p>You are currently viewing {$page.params.namespace}</p>
-  {/if}
-</div>
+  $: {
+    if (show) {
+      const focusable = getFocusableElements();
+      const inputElement = focusable[0];
+      inputElement.focus();
+    }
+  }
+</script>
 
-<div class="mb-5 flex rounded-md border border-gray-900 p-1 pr-4" role="search">
-  <div class="ml-4 mr-2">
-    <Icon name="search" />
+<svelte:window on:keydown|stopPropagation={handleKeyboardNavigation} />
+<div
+  class="h-[400px] w-[300px] overflow-auto py-4 px-12 md:w-[500px]"
```

Comment:
Some positioning things we might want to consider:

1. What the namespace list looks like on smaller breakpoints
![Screenshot 2023-03-16 at 1 09 54 PM](https://user-images.githubusercontent.com/15069288/225730850-307c7fff-1376-4c6a-916b-f9d43ec974a0.png)

2. If the data encoder help text on hover should show up over the user account info.
![Screenshot 2023-03-16 at 1 11 06 PM](https://user-images.githubusercontent.com/15069288/225730801-2d781fa2-6772-46ee-a707-8d8c148f4033.png)

Reply:
1. I made better responsive max widths for the dropdown and truncate long namespace names on sm screens.

<img width="245" alt="Screen Shot 2023-03-16 at 3 23 55 PM" src="https://user-images.githubusercontent.com/7967403/225744614-39629b2c-129b-4965-9cc8-f59b4f88a243.png">

2. I think we would keep the user menu over the tooltip.